File: src/main/java/vocadb/notification/reader/client/query/OptionalFields.java
package vocadb.notification.reader.client.query;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum OptionalFields {
    @JsonProperty("None")
    NONE,

    @JsonProperty("Albums")
    ALBUMS,
    @JsonProperty("Artists")
    ARTISTS,
    @JsonProperty("Description")
    DESCRIPTION,
    @JsonProperty("Groups")
    GROUPS,
    @JsonProperty("Lyrics")
    LYRICS,
    @JsonProperty("KnownLanguages")
    KNOWN_LANGUAGES,
    @JsonProperty("MainPicture")
    MAIN_PICTURE,
    @JsonProperty("Members")
    MEMBERS,
    @JsonProperty("Names")
    NAMES,
    @JsonProperty("OldUsernames")
    OLD_USERNAMES,
    @JsonProperty("PVs")
    PVS,
    @JsonProperty("Tags")
    TAGS,
    @JsonProperty("ThumbUrl")
    THUMB_URL,
    @JsonProperty("WebLinks")
    WEB_LINKS;

}

File: src/main/java/vocadb/notification/reader/client/api/enums/OptionalFields.kt
package vocadb.notification.reader.client.api.enums

import com.fasterxml.jackson.annotation.JsonValue

enum class OptionalFields(@JsonValue val value: String) {
    NONE("None"),

    ALBUMS("Albums"),
    ARTISTS("Artists"),
    DESCRIPTION("Description"),
    GROUPS("Groups"),
    KNOWN_LANGUAGES("KnownLanguages"),
    MAIN_PICTURE("MainPicture"),
    MEMBERS("Members"),
    NAMES("Names"),
    OLD_USERNAMES("OldUsernames"),
    PVS("PVs"),
    TAGS("Tags"),
    THUMB_URL("ThumbUrl"),
    WEB_LINKS("WebLinks");
}
